rmw_fastrtps_shared_cpp/src/rmw_wait.cpp

```diff
@@ -114,89 +55,94 @@ __rmw_wait(
   // error.
   // - Heap is corrupt.
   // In all three cases, it's better if this crashes soon enough.
-  auto wait_set_info = static_cast<CustomWaitsetInfo *>(wait_set->data);
-  std::mutex * conditionMutex = &wait_set_info->condition_mutex;
-  std::condition_variable * conditionVariable = &wait_set_info->condition;
+  auto fastdds_wait_set = static_cast<eprosima::fastdds::dds::WaitSet *>(wait_set->data);
+  bool no_has_wait = false;
```

Comment:
I would name this differently. Something like `skip_wait`.

Reply:
Done in dc5d78db